File: server/src/main/java/com/bettercallxiaojin/home/pojo/entity/Organization.java
package com.bettercallxiaojin.home.pojo.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Organization {
    private String id;
    private String userId;
    private String type;
    private String name;
    private LocalDate startDate;
    private LocalDate endDate;
    private String position;
    private String description;
    private String location;
}

File: web/src/models/organization.ts
// 请求类型
export interface OrganizationRequest {
  id: string;
  type?: string;
  name?: string;
  startDate?: string;
  endDate?: string;
  position?: string;
  description?: string;
  location: string;
}

export interface AddOrganizationRequest {
  type: string;
  name: string;
  startDate: string;
  endDate?: string;
  position: string;
  description?: string;
  location?: string;
}

// 响应数据类型
export interface OrganizationData {
  id: string;
  type: string;
  name: string;
  startDate: string; // LocalDate → string
  endDate: string;   // LocalDate → string
  position: string;
  description: string;
  location: string;
}